test.md

```diff
@@ -10,24 +10,34 @@ module WASM-TEST
     imports WASM
 ```
 
+Auxiliary
+---------
+
+Here we extend the sort `Stmt` by adding a new subsort called `Auxil`.
+This subsort contains Auxiliary functions that only used in our KWASM semantics but not in the official specification including assertions, initializing a global variable and the invocation of a function by exported name.
+
+```k
+    syntax Stmt ::= Auxil
+```
+
 Assertions
 ----------
 
 These assertions will check the supplied property, and then clear that state from the configuration.
 In this way, tests can be written as a serious of setup, execute, assert cycles which leaves the configuration empty on success.
 
-We'll make `Assertion` a subsort of `Instr`, so that we can easily consume `Instr` with `trap`s, but not consume `Assertion`s.
+We'll make `Assertion` a subsort of `Auxil`, so that we can easily consume `Instr` with `trap`s, but not consume `Assertion`s.
 This will allow `trap` to "bubble up" (more correctly, to "consume the continuation") until it reaches its paired `#assertTrap_` statement.
 
 ```k
-    syntax Instr ::= Assertion
+    syntax Auxil ::= Assertion
  // --------------------------
     rule <k> trap ~> (L:Label => .) ... </k>
     rule <k> trap ~> (F:Frame => .) ... </k>
     rule <k> trap ~> (.Instrs => .) ... </k>
-    rule <k> trap ~> (I:Instr => .) ... </k> requires notBool isAssertion(I)
+    rule <k> trap ~> (I:Instr => .) ... </k>
 
-    rule <k> trap ~> (I:Instr IS:Instrs => I ~> IS) ... </k>
+    rule <k> trap ~> (S:Stmt SS:Stmts => S ~> SS) ... </k>
```

Comment:
If a module is being defined and a `trap` occurs, the trap should propagate all the way to the top level. From the official test:
```
(assert_trap
  (module (func $main (unreachable)) (start $main))
  "unreachable"
)
```